tools/mafft/mafft.xml

```diff
@@ -65,36 +82,76 @@
             <when value="mafft-linsi"/>
             <when value="mafft-qinsi"/>
             <when value="mafft-xinsi"/>
+            <when value="custom">
+                <param name="distance_method" type="select" display="radio" label="Distance method" help="Distance method must be chosen regarding your data">
+                    <option value="6merpair" selected="true">Shared 6mers distance (fastest)</option>
+                    <option value="globalpair">Global alignment (NW)</option>
+                    <option value="localpair">Local alignment (SW)</option>
+                    <option value="genafpair">Local, affine gap cost</option>
+                </param>
+                <param name="ep" type="float" value="0.123" label="Gap extend penalty" help="0.123 default value" />
+                <param name="retree" type="integer" value="2" min="1" max="100" label="Guide tree is built number times in the progressive stage." help="Valid with 6mer distance" />
+                <param name="iterations" type="integer" value="1" min="0" max="1000" label="Maximum number of iterations" help="1000 for maximum quality" />
+            </when>
         </conditional>
+        <param name="op" type="float" value="1.53" label="Gap opening penalty" help="1.53 default value" />
         <param name="adjustdirection" type="select" display="radio" label="Direction of nucleotide sequences" help="Generate reverse complement sequences, as necessary, and align them together with the remaining sequences">
             <option value="--adjustdirection">adjust direction</option>
             <option value=" " selected="true">do not adjust direction</option>
-            </param>
+        </param>
+        <conditional name="matrix_condition">
+            <param name="use_matrix" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Use matrix ?" help="Usefull only for amino acids" />
```

Comment:
Can we change this to a select box, please. 
This is recommended by the current best practice guide: http://galaxy-iuc-standards.readthedocs.org/en/latest/best_practices/tool_xml.html?highlight=conditionals#booleans